format/CMakeLists.txt

```
@@ -0,0 +1,67 @@
+set(did_search ON)
+if(NOT DEFINED CLANG_FORMAT)
+    message(STATUS "Searching for VS clang-format")
+    set(did_search OFF)
+endif()
+
+if(ERROR_ON_CLANG_FORMAT_NOT_FOUND)
+    set(message_level FATAL_ERROR)
+else()
+    set(message_level WARNING)
+endif()
+
+find_program(CLANG_FORMAT
+    NAMES clang-format
+    PATHS "C:/Program Files/Microsoft Visual Studio/2022/Preview/VC/Tools/Llvm/bin"

```

Can we use the 64-bit hosted tool instead? (I'm aware it doesn't really matter for clang-format as opposed to the actual compiler, but I believe it's simpler to default everything to x64.)
```suggestion
    PATHS "C:/Program Files/Microsoft Visual Studio/2022/Preview/VC/Tools/Llvm/x64/bin"
```

---

Is there a reason to do this? AFAICT this just makes it impossible to build on x86-windows and doesn't gain anything (i mean, I guess not that we care about x86-windows host machines)

---

While I think it's a feature and not a bug to require 64-bit machines for building the STL :joy_cat:, I have no objections to using the 32-bit hosted clang-format as there are no OOM considerations. (I do note that 32-bit binaries are less secure, but this also has no impact on clang-format.)